.changeset/happy-dreams-sing.md

```diff
@@ -0,0 +1,26 @@
+---
+"@fluidframework/sequence": minor
+---
+
+1. Deprecate API `findOverlappingIntervals` and `gatherIterationResults` from `IntervalCollection`, these functionalities are moved to the `OverlappingIntervalsIndex`. Users are advised to independently attach the index to the collection and utilize the API accordingly, for instance:
+
+```typescript
+const overlappingIntervalsIndex = createOverlappingIntervalsIndex(client, helpers);
+collection.attachIndex(overlappingIntervalsIndex)
+const result1 = overlappingIntervalsIndex.findOverlappingIntervals(start, end);
+
+const result2 = [];
+overlappingIntervalsIndex.gatherIterationResults(result2, true);
+```
+
+2. Deprecate API `previousInterval` and `nextInterval` from `IntervalCollection`, these functionalities are moved to the `OverlappingIntervalsIndex`. Users are advised to independently attach the index to the collection and utilize the API accordingly, for instance:
```

Comment:
looks like these are on the EndpointInRange index, not overlapping. also, they're not exported by the interface (but they should be for this guidance to be true).

Reply:
yes this is a mistake, they should be on the `EndpointIndex` (potentially this functionality can be combined into `EndpointInRangeIndex` as well). Actually the `IEndpointIndex` and `createEndpointIndex` are already exported, so I think this guidance seems accurate..?

Reply:
yeah, I was just looking at local contents. The API diff in this PR looks good once you fix the guidance to partners here.